paddle/phi/core/dense_tensor.cc

```diff
@@ -107,20 +108,27 @@ void* DenseTensor::AllocateFrom(Allocator* allocator,
     VLOG(10) << "change data type in mutbale_data, target dtype - " << dtype;
     meta_.dtype = dtype;
   }
-  PADDLE_ENFORCE(
-      valid(),
-      phi::errors::PreconditionNotMet(
-          "The meta data must be valid when call the mutable data function."));
+
   size_t bytes = numel() * SizeOf(this->dtype());
   if (requested_size) {
-    PADDLE_ENFORCE_GE(requested_size,
-                      bytes,
-                      phi::errors::InvalidArgument(
-                          "The reserved size %d should be enough to meet the "
-                          "volume required by metadata %d.",
-                          requested_size,
-                          bytes));
+    if (check_size) {
+      PADDLE_ENFORCE(
+          valid(),
+          phi::errors::PreconditionNotMet("The meta data must be valid when "
+                                          "call the mutable data function."));
+      PADDLE_ENFORCE_GE(requested_size,
+                        bytes,
+                        phi::errors::InvalidArgument(
+                            "The reserved size %d should be enough to meet the "
+                            "volume required by metadata %d.",
+                            requested_size,
+                            bytes));
+    }
     bytes = requested_size;
+  } else {
+    if (!check_size) {
```

Comment:
Counterintuitive code here. Maybe `check_size` can be renamed with `allocate_zero_size` or `fack_allocate`, and directly assign `0` to `bytes` when passing this parameter?

Reply:
good suggestion, thx.